core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -806,6 +806,22 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     }
     // Clean the blacklist from the expired entries.
     clearBlacklist(CLEAN_INTERVAL_S)
+
+    // Delete driver logs from the configured spark dfs dir that exceed the configured max age
+    try {
+      val hdfsDir = conf.get("spark.driver.log.dfsDir")
+      val appDirs = fs.listLocatedStatus(new Path(hdfsDir))
+      while (appDirs.hasNext()) {
+        val appDirStatus = appDirs.next()
+        if (appDirStatus.getModificationTime() < maxTime) {
+          logInfo(s"Deleting expired driver log for: ${appDirStatus.getPath().getName()}")
+          deleteLog(appDirStatus.getPath())
+        }
+      }
+    } catch {
+      case nse: NoSuchElementException => // no-op
```

Comment:
What call throws this exception?

Reply:
conf.get("spark.driver.log.dfsDir")

Reply:
It's an optional config. If you use the config constant this will go away. (And even in the legacy case you should use `conf.getOption` in those cases.)